go/analysis/passes/printf/printf.go

```diff
@@ -66,6 +68,51 @@ argument list. Otherwise it is assumed to be Print-like, taking a list
 of arguments with no format string.
 `
 
+type PrintKind int
+
+const (
+	// Not a print or printf func or wrapper
+	KindNo PrintKind = iota
+	// Print or Print wrapper
+	KindPrint
+	// Printf or Printf wrapper
+	KindPrintf
+)
+
+func (w PrintKind) String() string {
+	switch w {
+	case KindPrint:
+		return "printWrapper"
+	case KindPrintf:
+		return "printfWrapper"
+	}
+	return ""
+}
+
+// Result is the printf analyzer's result type. It allows checking if
+// a function is a print or printf wrapper.
+type Result struct {
+	set map[*types.Func]PrintKind
+}
+
+// IsPrint returns
+func (r *Result) IsPrint(fn *types.Func) PrintKind {
```

Comment:
"returns..."?

Call this method Kind, since it returns the wrapper kind.

// Kind reports whether fn is a wrapper around fmt.Print or fmt.Printf.
func (*Result) Kind(fn *types.Func) Kind {